src/lib/remark-plugins/rewrite-static-tutorials-assets/index.ts

```diff
@@ -25,13 +25,23 @@ import { Image, Definition } from 'mdast'
 const ASSET_API_ENDPOINT =
 	process.env.ASSET_API_ENDPOINT || `${process.env.MKTG_CONTENT_API}/api/assets`
 
+/**
+ * This Plugin rewrites src asset paths for tutorials content. With tutorials, images live in a
+ * /public directory in the tutorials repository.
+ *
+ * For dev portal previews / prod, we source these image paths from the mktg-content-api,
+ * which uses the GitHub API. For tutorials repo previews, we use the PREVIEW_BRANCH env
+ * to target the correct path via the GitHub API.
+ *
+ * For authors working on content locally, we spin up a custom asset server within docker
+ * and the paths are served 1-1.
+ */
 export const rewriteStaticAssetsPlugin: Plugin = () => {
```

Comment:
💭 perhaps should add some tests for this plugin 😅

Reply:
Good thought 😅 — wanna create & link-in-a-comment an asana task? 🙏

Reply:
https://app.asana.com/0/1202097197789424/1204921235104809